code/modules/clothing/shoes/marine_shoes.dm

```diff
@@ -12,40 +12,51 @@
 	min_cold_protection_temperature = SHOE_MIN_COLD_PROTECTION_TEMPERATURE
 	max_heat_protection_temperature = SHOE_MAX_HEAT_PROTECTION_TEMPERATURE
 	siemens_coefficient = 0.7
-	var/obj/item/knife
-
-/obj/item/clothing/shoes/marine/Destroy()
-	if(knife)
-		QDEL_NULL(knife)
-	return ..()
+	var/obj/item/storage/internal/pockets = /obj/item/storage/internal/shoes/boot_knife
+
+/obj/item/storage/internal/shoes/boot_knife
+	max_storage_space = 3
+	storage_slots = 1
+	can_hold = list(
+		/obj/item/weapon/combat_knife,
+		/obj/item/attachable/bayonetknife,
+		/obj/item/weapon/throwing_knife
+	)
+
+/obj/item/clothing/shoes/marine/Initialize()
+	. = ..()
+	pockets = new pockets(src)
 
-//ATTACK HAND IGNORING PARENT RETURN VALUE
 /obj/item/clothing/shoes/marine/attack_hand(mob/living/user)
-	if(knife && src.loc == user && !user.incapacitated()) //Only allow someone to take out the knife if it's being worn or held. So you can pick them up off the floor
-		if(user.put_in_active_hand(knife))
-			to_chat(user, "<span class='notice'>You slide [knife] out of [src].</span>")
-			playsound(user, 'sound/weapons/guns/interact/shotgun_shell_insert.ogg', 15, 1)
-			knife = 0
-			update_icon()
-	else
+	if(!pockets)
```

Comment:
Should this ever lack pockets?
Also, shouldn't the click only transfer when held? You don't want to click on the container when trying to grab the boots from the ground.

Reply:
I removed the redundancy check.
It already requires you to click and hold to you if you need to get into the internal inventory that way if that's what you mean.